### PR TITLE
CRM: Resolves #3330 - use "task" term instead of "event"

### PR DESCRIPTION
**Pull request by tbradsha**
CRM: Resolves Automattic/zero-bs-crm#3330 - use task instead of event

## Proposed changes:

This PR updates many references to use `task` instead of `event` wherever possible (just shy of 700 references).

**Note:** please ignore the one failing PHPCS check titled `Linting / PHP Code Sniffer (changes to excluded files only) (pull_request)`. We will skip this prior to merge; instructions to do so can be found in p1695240852475789-slack-CDLH4C1UZ.

Things that did NOT change:

* DAL (with the exception of constant updates) and DAL helper functions
* Database
* API endpoint names (i.e. `create_event` and `events`)
* Permission capabilities (e.g. `admin_zerobs_view_events`)
* Hard-coded low-level references to object type as 'event' or `zerobs_event` (e.g. in the URL 😢)
* Things that rely on dynamically-accessed globals (e.g. `$zeroBSCRM_columns_event`)
* Things that rely on dynamically-accessed JS vars (e.g. `zeroBSCRMJS_listView_event_bulkActionTitle_markcomplete`)
* Functions and event names in the old internal automator system

Of note, `ZBS_TYPE_TASK` now exists and has replaced `ZBS_TYPE_EVENT` throughout the codebase. The old constant still remains for backward-compatibility.

One can search for references with the following regex:

`\b(?!(pr|admin_zerobs_|addUpdate))\w*event(?!(s->|.prevent|Listener|ually))`

And exclude the following:

`js/lib/*, sass/semantic-ui/*, changelog/*, */LICENSE, licenses/*, *.txt, *.md, *.lock, *.dist, *.pot`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Make sure everything task-related runs as before. Tests should pass. The new Automation stuff should be properly renamed.

### CODE REVIEW

**Comment by cleacos** [comment]
The test linting (PHP Code Sniffer) continue failing

**Comment by tbradsha** [comment]
> The test linting (PHP Code Sniffer) continue failing

Yes, this is expected. We will skip the check once approved/ready to merge.

**Comment by cleacos** [comment]
Let's wait to have all the Automation changes merged before re-reviewing this PR and fixing the conflicts.

File: projects/plugins/crm/includes/ZeroBSCRM.AJAX.php [hunk 1]
```diff
@@ -3488,7 +3488,7 @@ function zeroBSCRM_AJAX_listViewRetrieveData() {
 
 			/*
 			==============================================================================
-			===================== EVENT ================================================= */
+			===================== TAASK ================================================= */
```
Comment (coder-karen) [inline projects/plugins/crm/includes/ZeroBSCRM.AJAX.php]:
```suggestion
			===================== TASK ================================================= */
```

**Review by coder-karen — COMMENTED** [review]
Looks good to me! Tests all still pass, functionality works. Spotted one small comment typo, but I can add a commit with that changed.

**Review by coder-karen — APPROVED** [review]

**Review by cleacos — APPROVED** [review]
I tested it as well: 

- PHPUnit tests passed.
- Acceptance tests passed.

I see many linter issues in the GH checks...

image

File: projects/plugins/crm/includes/ZeroBSCRM.AJAX.php (hunk 1 above)
Comment (tbradsha) [reply projects/plugins/crm/includes/ZeroBSCRM.AJAX.php]: > Spotted one small comment typo

Written by a sheep!

**Pull request merged**